netutils/ping.py

```diff
@@ -25,7 +25,10 @@ def tcp_ping(ip: str, port: int, timeout: int = 1) -> bool:  # pylint: disable=i
     sckt.settimeout(int(timeout))
     try:
         sckt.connect((ip, int(port)))  # pylint: disable=invalid-name
-        sckt.shutdown(int(timeout))
+        sckt.shutdown(socket.SHUT_RDWR)
         return True
-    except socket.error:
+    # We really only want to know if the TCP connection timed out
+    # or if there was an error resolving the DNS name, if anything
+    # else has happened the error should be raised.
+    except (socket.timeout, socket.gaierror):
```

Comment:
Please add the finally on this to always disconnect

Reply:
I tried that, that ends up raising a new exception because if the connection failed you can't actually close it.

Reply:
wondering if we could save ourselves a bit and use `connect_ex()` instead? https://docs.python.org/3.7/library/socket.html#socket.socket.connect_ex

Reply:
I'm not sure if that changes the logic required, though. You still end up checking of errno is zero, which would be the same as not catching the exception.

Reply:
good point

Reply:
I initially thought that `socket.gaierror` made sense, but I am second guessing that. I think it is best to raise a stacktrace in that instance, since the error should be different for timeout vs a bad dns name.

Reply:
okay, `finally` added with `close()` and `socket.gaierror` has been removed.